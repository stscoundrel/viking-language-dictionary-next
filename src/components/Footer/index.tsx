import ExternalLink from 'components/ExternalLink'
import LetterLink from 'components/LetterLink'
import ContentArea from 'components/ContentArea'
import { AlphabetLetter } from 'lib/services/dictionary'
import styles from './Footer.module.scss'

interface FooterProps{
  letters: AlphabetLetter[]
}

export default function Footer({ letters }: FooterProps) {
  return (
    <footer className={styles.section}>
      <div className="container">

        <ContentArea>
          <h2>About</h2>
          <p><em>`Viking Language`</em> is a book series by
          <ExternalLink
            title="Jesse L. Byock"
            href="http://www.viking.ucla.edu/"
          />
          which teaches basics of Old Norse language.<br />
          The books are accompanied by a small dictionary at the end,
          providing learners a place to look for translations of the books passages.</p>

          <p>This searchable dictionary is compiled from a
          <ExternalLink
              title="blog post"
              href="https://oldnorse.org/2020/09/06/the-old-norse-dictionary/"
            />
           by the books publisher
          <ExternalLink
              title="Jules William Press"
              href="https://juleswilliampress.com/"
            />
          </p>

          <p>All rights to Viking Language series belong to
            Jules William Press and Jesse L. Byock.
            Data in this dictionary is based on data made openly available by the publisher.
          </p>
        </ContentArea>

        <ContentArea>
          <h3>Old Norse language</h3>
          <p>Old Norse was a North Germanic language that was spoken by inhabitants of
          Scandinavia and their overseas settlements from about the 7th to the 15th centuries.</p>

          <p>Also known as &quot;the viking language&quot;,
          &quot;Old Nordic&quot;, or
          &quot;Old Scandinavian&quot;</p>
        </ContentArea>

        <div className={styles.navs}>
          <nav className={styles.nav}>
            <h4 className={styles.navTitle}>Dictionary project</h4>
            <ul>
              <li>
                <ExternalLink
                  title="Source code"
                  href="https://github.com/stscoundrel/viking-language-dictionary-next"
                />
              </li>
              <li>
                <ExternalLink
                  title="Data source"
                  href="https://github.com/stscoundrel/viking-language-dictionary"
                />
              </li>
              <li>
                <ExternalLink
                  title="Abbreviations"
                  href="https://github.com/stscoundrel/viking-language-dictionary-abbrevations"
                />
              </li>
            </ul>
          </nav>

          <nav className={styles.nav}>
            <h4 className={styles.navTitle}>Related dictionary projects</h4>
            <ul>
              <li>
                <ExternalLink
                  title="Cleasby and Vigfusson Old Norse Dictionary"
                  href="https://cleasby-vigfusson-dictionary.vercel.app/"
                />
              </li>
              <li>
                <ExternalLink
                  title="A Concise Dictionary of Old Icelandic"
                  href="https://old-icelandic.vercel.app/"
                />
              </li>
              <li>
                <ExternalLink
                  title="Dictionary of the Old Norwegian Language"
                  href="https://viking-language-dictionary.vercel.app/"
                />
              </li>
              <li>
                <ExternalLink
                  title="Old Norse Alphabet"
                  href="https://github.com/stscoundrel/old-norse-alphabet"
                />
              </li>
            </ul>
          </nav>

          <nav className={styles.nav}>
            <h4 className={styles.navTitle}>Quick links</h4>
            <ul className={styles.navColumns}>
              {letters.map((entry) => (
                <li className={styles.navColumnItem} key={entry.slug}>
                  <LetterLink letter={entry} />
                </li>
              ))}
            </ul>
          </nav>
        </div>
        <small className={styles.copyright}>{`Copyright Â© 2022 - ${new Date().getFullYear()}`}
          <br />
          <ExternalLink
            title="Sampo Silvennoinen / StScoundrel"
            href="https://github.com/stscoundrel"
          />
        </small>
      </div>
    </footer>
  )
}
