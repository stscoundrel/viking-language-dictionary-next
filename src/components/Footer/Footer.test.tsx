import ReactDOM from 'react-dom/client'
import renderer from 'react-test-renderer'
import Footer from './index'

describe('Footer component', () => {
  test('Does not crash', () => {
    const div = document.createElement('div')
    const root = ReactDOM.createRoot(div)
    root.render(<Footer />)
  })

  test('Matches snapshot', () => {
    const tree = renderer.create(<Footer />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
