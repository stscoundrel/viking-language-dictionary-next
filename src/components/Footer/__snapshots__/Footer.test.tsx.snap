// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Matches snapshot 1`] = `
<footer
  className="section"
>
  <div
    className="container"
  >
    <section
      className="section"
    >
      <h2>
        About
      </h2>
      <p>
        \`"
        <em>
          Viking Language\`
        </em>
        " is a book series by
        <a
          href="http://www.viking.ucla.edu/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Jesse L. Byock
        </a>
        which teaches basics of Old Norse language.
        <br />
        The books are accompanied by a small dictionary at the end, providing learners a place to look for translations of the books passages.
      </p>
      <p>
        This searchable dictionary is compiled from a
        <a
          href="https://oldnorse.org/2020/09/06/the-old-norse-dictionary/"
          rel="noopener noreferrer"
          target="_blank"
        >
          blog post
        </a>
        by the books publisher
        <a
          href="https://juleswilliampress.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Jules William Press
        </a>
      </p>
      <p>
        All rights to Viking Language series belong to Jules William Press and Jesse L. Byock. Data in this dictionary is based on data made openly available by the publisher.
      </p>
    </section>
    <section
      className="section"
    >
      <h3>
        Old Norse language
      </h3>
      <p>
        Old Norse was a North Germanic language that was spoken by inhabitants of Scandinavia and their overseas settlements from about the 7th to the 15th centuries.
      </p>
      <p>
        Also known as "the viking language", "Old Nordic", or "Old Scandinavian"
      </p>
    </section>
    <div
      className="navs"
    >
      <nav
        className="nav"
      >
        <h4
          className="navTitle"
        >
          Dictionary project
        </h4>
        <ul>
          <li>
            <a
              href="https://github.com/stscoundrel/viking-language-dictionary-next"
              rel="noopener noreferrer"
              target="_blank"
            >
              Source code
            </a>
          </li>
          <li>
            <a
              href="https://github.com/stscoundrel/viking-language-dictionary"
              rel="noopener noreferrer"
              target="_blank"
            >
              Data source
            </a>
          </li>
          <li>
            <a
              href="https://github.com/stscoundrel/viking-language-dictionary-abbrevations"
              rel="noopener noreferrer"
              target="_blank"
            >
              Abbreviations
            </a>
          </li>
        </ul>
      </nav>
      <nav
        className="nav"
      >
        <h4
          className="navTitle"
        >
          Related dictionary projects
        </h4>
        <ul>
          <li>
            <a
              href="https://cleasby-vigfusson-dictionary.vercel.app/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Cleasby and Vigfusson Old Norse Dictionary
            </a>
          </li>
          <li>
            <a
              href="https://old-icelandic.vercel.app/"
              rel="noopener noreferrer"
              target="_blank"
            >
              A Concise Dictionary of Old Icelandic
            </a>
          </li>
          <li>
            <a
              href="https://old-norwegian-dictionary.vercel.app/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Dictionary of the Old Norwegian Language
            </a>
          </li>
          <li>
            <a
              href="https://github.com/stscoundrel/old-norse-alphabet"
              rel="noopener noreferrer"
              target="_blank"
            >
              Old Norse Alphabet
            </a>
          </li>
        </ul>
      </nav>
      <nav
        className="nav"
      >
        <h4
          className="navTitle"
        >
          Quick links
        </h4>
        <ul
          className="navColumns"
        >
          <li
            className="navColumnItem"
          >
            <a
              className="link"
              href="/letter/a"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              A
            </a>
          </li>
          <li
            className="navColumnItem"
          >
            <a
              className="link"
              href="/letter/b"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              B
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <small
      className="copyright"
    >
      Copyright Â© 2022 - 2021
      <br />
      <a
        href="https://github.com/stscoundrel"
        rel="noopener noreferrer"
        target="_blank"
      >
        Sampo Silvennoinen / StScoundrel
      </a>
    </small>
  </div>
</footer>
`;
